### ISSUE TITLE
send() to Iterator produces strange warning

### ISSUE THREAD

**Issue by jolaf** [agronholm/typeguard#77]
The following code:
```python3
from typing import Iterator

from typeguard import TypeChecker

def f() -> Iterator[str]:
    yield "OK"

with TypeChecker('__main__'):
    print(f().send(None))
```
works fine, but produces the following warning from `mypy`:
```python3
Test.py:9:11: error: "Iterator[str]" has no attribute "send"
```
which is understandable, but also produces a weird warning from `typeguard`:
```python3
/usr/local/lib/python3.6/dist-packages/typeguard/__init__.py:916: TypeWarning: [MainThread] return from __main__.f() at Test.py:6: type of yielded value must be collections.abc.Iterator; got str instead
```
It looks to me like the warning itself is needed, but the "must" and "got" types are wrong.

**Comment by jolaf** [comment]
Another issue that seems related to this one, looks difficult to reproduce.
It occurs in one point in my application only once, all the other times the code runs through that point, it produces no warning.
All my attempts to separate the issue and create compact test failed.

So I provide what I have hoping it would help.

The code similar to this one:
```python3
from typing import Generator, Iterator

from typeguard import TypeChecker

class A:
    def f(self) -> Iterator[str]:
        yield "OK"

def g() -> Generator[Generator[str, None, None], None, None]:
    yield A().f()

with TypeChecker('__main__'):
    print(next(next(g())))
```
sometimes causes a warning like this:
```python3
/usr/local/lib/python3.6/dist-packages/typeguard/__init__.py:916: TypeWarning: [MainThread] return from __main__.g() at Test.py:10: type of yielded value must be str; got generator instead
```

**Comment by agronholm** [comment]
This reminds me, I need to add `typing.Iterator` to the list of type checked annotations.
